src/core/Form/SearchInput/SearchInput.tsx

```diff
@@ -228,43 +231,57 @@ class BaseSearchInput extends Component<SearchInputProps> {
         <LabelText htmlFor={this.id} labelMode={labelMode} as="label">
           {labelText}
         </LabelText>
-        <HtmlDiv className={searchInputClassNames.functionalityContainer}>
-          <HtmlDiv className={searchInputClassNames.inputElementContainer}>
-            <HtmlInputWithRef
-              {...passProps}
-              {...getDescribedBy()}
-              forwardRef={this.inputRef}
-              aria-invalid={status === 'error'}
-              id={this.id}
-              className={searchInputClassNames.inputElement}
-              type="search"
-              role="searchbox"
-              value={this.state.value}
-              placeholder={visualPlaceholder}
-              onChange={(event: ChangeEvent<HTMLInputElement>) => {
-                conditionalSetState(event.currentTarget.value);
-              }}
-              onKeyPress={onKeyPress}
-              onKeyDown={onKeyDown}
-            />
-          </HtmlDiv>
-          <HtmlButton {...clearButtonProps}>
-            <VisuallyHidden>{clearButtonLabel}</VisuallyHidden>
-            <Icon
-              aria-hidden={true}
-              icon="close"
-              className={searchInputClassNames.clearIcon}
-            />
-          </HtmlButton>
-          <HtmlButton {...searchButtonDerivedProps}>
-            <VisuallyHidden>{searchButtonLabel}</VisuallyHidden>
-            <Icon
-              aria-hidden={true}
-              icon="search"
-              className={searchInputClassNames.searchIcon}
-            />
-          </HtmlButton>
-        </HtmlDiv>
+        <Debounce waitFor={this.props.debounce || 0}>
```

Comment:
Wasn't there discussion that there is no need to tell the time as zero, as undefined would be handled by the Debounce util?

So the use will be like:
`<Debounce waitFor={this.props.debounce}>`

Reply:
Truthy check should be sufficient if used in Debounce component. !!waitFor should cover all cases where waitFor is 0, "", undefined or null => not present and can be handled the same way.

Reply:
In addition, I would either put Debounce as high up in the component hierarchy as possible (e.g. wrap the exported SearchInput) or right next to the component it is used for, in this case wrap just the functionalityContainer div since components outside it have no use for the functionality.

Reply:
Ah good points. I'll fix the waitFor value handling. At this point in the year everything that's not immediately written down will be forgotten 😅. What comes to the placement, to me it seems it's in the right spot. Wrapping the functionalityContainer, that is.